services/api/src/resources/project/harborSetup.js

```diff
@@ -24,133 +25,293 @@ const lagoonWebhookAddress = R.compose(
-      } else {
-        logger.error(`Unable to get the harbor project id of "${lagoonProjectName}" !!`)
+async function removeRobot(sqlClient /* : MariaSQL */, harborClient /* : harborClient */, lagoonProjectName /* : string */, harborProjectID /* : string */, robotName /* : string */) {
+  // Returns false on an error and true on a success
+
+  // Check for preexisting robot account of the same name
+  var robots = []
+  try {
+    const res = await harborClient.get(`projects/${harborProjectID}/robots`);
+
+    for (var i = 0; i < res.body.length; i++) {
+      if (res.body[i].name == `robot\$${robotName}`) {
+        robots.push(res.body[i])
       }
     }
-    logger.debug(`Harbor project id for ${lagoonProjectName}: ${harborProjectID}`)
+  } catch (err) {
+    logger.info(`Unable to retrieve list of current robot accounts for Harbor Project: ${lagoonProjectName}`, err)
+  }
 
-    // Create robot account for new harbor project
-    try {
-      const res = await harborClient.post(`projects/${harborProjectID}/robots`, {
+  // Remove preexisting robot account
+  try {
+    for (var i = 0; i < robotNames.length; i++) {
+      var result = await harborClient.delete(`projects/${harborProjectID}/robots/${robots[i].id}`, {
         body: {
-          name: lagoonProjectName,
-          access: [
-            {
-              resource: `/project/${harborProjectID}/repository`,
-              action: "push"
-            }
-          ]
+          project_id: harborProjectID,
+          robot_id: robots[i].id
         }
-      })
-      var harborTokenInfo = res.body
-      logger.debug(`Robot was created for Harbor project ${lagoonProjectName} !`)
-    } catch (err) {
-      // 409 means project already exists
-      // 201 means project created successfully
-      if (err.statusCode == 409) {
-        logger.warn(`Unable to create a robot account for harbor project "${lagoonProjectName}", as a robot account of the same name already exists!`)
-      } else {
-        logger.warn(`Unable to create a robot account for harbor project "${lagoonProjectName}"!`, err)
-      }
+      });
+      logger.debug(`Removed harbor robot account: ${robots[i].name}`)
     }
+  } catch (err) {
+    logger.error(`Unable to remove pre-existing robot accounts for Harbor project: ${lagoonProjectName}`, err)
+    return false
+  }
+  return true
+}
 
-    // Set Harbor env vars for lagoon environment
-    try {
-      await query(
-        sqlClient,
-        Sql.insertEnvVariable({
-          "name": "INTERNAL_REGISTRY_URL",
-          "value": lagoonHarborRoute,
-          "scope": "INTERNAL_CONTAINER_REGISTRY",
-          "project": projectID,
-        }),
-      );
-      logger.debug(`Environment variable INTERNAL_REGISTRY_URL for ${lagoonProjectName} created!`)
+async function removeAllRobots(sqlClient /* : MariaSQL */, harborClient /* : harborClient */, lagoonProjectName /* : string */, harborProjectID /* : string */) {
```

Comment:
this code is never used? I guess it can be removed then?

Reply:
Eventually, we will need to handle the removal of older robot accounts; this function will be relevant when we want to implement that removal functionality.

Reply:
okay, but then please let's not ship code that is not used anywhere, remove the code from this PR, and add it to another branch that we can then use again when the time is right for the removal functionality.

Reply:
Understood. Function removed and stored in the `add_harbor_robot_removal` branch.